csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -170,12 +170,33 @@ public void AppendExecutionProvider_Nuphar(string settings = "")
         #endregion //ExecutionProviderAppends
 
         #region Public Methods
+
+        /// <summary>
+        /// (Deprecated) Loads a DLL named 'libraryPath' and looks for this entry point:
+        /// OrtStatus* RegisterCustomOps(OrtSessionOptions* options, const OrtApiBase* api);
+        /// It then passes in the provided session options to this function along with the api base.
+        /// Deprecated in favor of RegisterCustomOpLibraryV2() because it provides users with the library handle 
+        /// to release when all sessions relying on it are destroyed
+        /// </summary>
         public void RegisterCustomOpLibrary(string libraryPath)
         {
             IntPtr libraryHandle = IntPtr.Zero;
             NativeApiStatus.VerifySuccess(NativeMethods.OrtRegisterCustomOpsLibrary(handle, libraryPath, out libraryHandle));
         }
 
+        /// <summary>
+        /// Loads a DLL named 'libraryPath' and looks for this entry point:
+        /// OrtStatus* RegisterCustomOps(OrtSessionOptions* options, const OrtApiBase* api);
+        /// It then passes in the provided session options to this function along with the api base.
+        /// The handle to the loaded library is returned in 'libraryHandle'. 
+        /// It can be freed by the caller after all sessions using the passed in
```

Comment:
>freed by the caller [](start = 22, length = 19)

The caller may think it needs a memory allocator to free it. It should be unloaded and we need to specify the API to do it. Otherwise, it is a GH issue.

Reply:
Why do we need an API to free a library handle ? We have not exposed anything in the C API to do this. We rely on FreeLibrary() on Windows to do that. The user can P/Invoke to free the handle. This is what the test does. Why expose in C# something that we explicitly decided not to expose in the C API ?